File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/BankContract.java
package org.mifos.mobilewallet.mifospay.bank;

import org.mifos.mobilewallet.core.domain.model.BankAccountDetails;
import org.mifos.mobilewallet.mifospay.base.BasePresenter;
import org.mifos.mobilewallet.mifospay.base.BaseView;

import java.util.List;

/**
 * Created by ankur on 09/July/2018
 */

public interface BankContract {

    interface BankAccountsPresenter extends BasePresenter {

        void fetchLinkedBankAccounts();
    }

    interface BankAccountsView extends BaseView<BankAccountsPresenter> {

        void showLinkedBankAccounts(List<BankAccountDetails> bankAccountList);
    }

    interface LinkBankAccountPresenter extends BasePresenter {

        void fetchBankAccountDetails(String bankName);
    }

    interface LinkBankAccountView extends BaseView<LinkBankAccountPresenter> {

        void addBankAccount(BankAccountDetails bankAccountDetails);
    }

    interface BankAccountDetailPresenter extends BasePresenter {

    }

    interface BankAccountDetailView extends BaseView<BankAccountDetailPresenter> {

    }


    interface DebitCardPresenter extends BasePresenter {

        void verifyDebitCard(String s, String s1, String s2);
    }

    interface DebitCardView extends BaseView<DebitCardPresenter> {

        void verifyDebitCardSuccess(String otp);

        void verifyDebitCardError(String message, int viewNumber);
    }

    interface UpiPinPresenter extends BasePresenter {

    }

    interface UpiPinView extends BaseView<UpiPinPresenter> {

    }

    interface SetupUpiPinPresenter extends BasePresenter {

        void setupUpiPin(BankAccountDetails bankAccountDetails, String upiPin);

        void requestOtp(BankAccountDetails bankAccountDetails);
    }

    interface SetupUpiPinView extends BaseView<SetupUpiPinPresenter> {

        void debitCardVerified(String otp);

        void setupUpiPinSuccess(String mSetupUpiPin);

        void setupUpiPinError(String message);
    }

}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/BankContract.kt
package org.mifos.mobilewallet.mifospay.bank

import org.mifos.mobilewallet.core.domain.model.BankAccountDetails
import org.mifos.mobilewallet.mifospay.base.BasePresenter
import org.mifos.mobilewallet.mifospay.base.BaseView

/**
 * Created by ankur on 09/July/2018
 */
interface BankContract {
    interface BankAccountsPresenter : BasePresenter {
        fun fetchLinkedBankAccounts()
    }

    interface BankAccountsView : BaseView<BankAccountsPresenter?> {
        fun showLinkedBankAccounts(bankAccountList: List<BankAccountDetails?>?)
    }

    interface LinkBankAccountPresenter : BasePresenter {
        fun fetchBankAccountDetails(bankName: String?)
    }

    interface LinkBankAccountView : BaseView<LinkBankAccountPresenter?> {
        fun addBankAccount(bankAccountDetails: BankAccountDetails?)
    }

    interface BankAccountDetailPresenter : BasePresenter
    interface BankAccountDetailView : BaseView<BankAccountDetailPresenter?>
    interface DebitCardPresenter : BasePresenter {
        fun verifyDebitCard(s: String?, s1: String?, s2: String?)
    }

    interface DebitCardView : BaseView<DebitCardPresenter?> {
        fun verifyDebitCardSuccess(otp: String?)
        fun verifyDebitCardError(message: String?)
    }

    interface UpiPinPresenter : BasePresenter
    interface UpiPinView : BaseView<UpiPinPresenter?>
    interface SetupUpiPinPresenter : BasePresenter {
        fun setupUpiPin(bankAccountDetails: BankAccountDetails?, upiPin: String?)
        fun requestOtp(bankAccountDetails: BankAccountDetails?)
    }

    interface SetupUpiPinView : BaseView<SetupUpiPinPresenter?> {
        fun debitCardVerified(otp: String?)
        fun setupUpiPinSuccess(mSetupUpiPin: String?)
        fun setupUpiPinError(message: String?)
    }
}